specs/wp-signup-spec.js

```diff
@@ -305,6 +305,216 @@ describe( `[${ host }] Sign Up  (${ screenSize }, ${ locale })`, function() {
+		} );
+
+		step( 'Can then enter account details', async function() {
+			const createYourAccountPage = await CreateYourAccountPage.Expect( driver );
+			await eyesHelper.eyesScreenshot( driver, eyes, 'Create Account Page' );
+			return await createYourAccountPage.enterAccountDetailsAndSubmit(
+				emailAddress,
+				blogName,
+				passwordForTestAccounts
+			);
+		} );
+
+		step(
+			'Can then see the sign up processing page which will automatically move along',
+			async function() {
+				if ( global.browserName === 'Internet Explorer' ) {
+					return;
+				}
+				const signupProcessingPage = await SignupProcessingPage.Expect( driver );
+				return await signupProcessingPage.waitToDisappear();
+			}
+		);
+
+		step(
+			'Can then see the secure payment page with the premium plan in the cart',
+			async function() {
+				const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+				await eyesHelper.eyesScreenshot( driver, eyes, 'Secure Payment Page' );
+				const premiumPlanInCart = await securePaymentComponent.containsPremiumPlan();
+				assert.strictEqual( premiumPlanInCart, true, "The cart doesn't contain the premium plan" );
+				const numberOfProductsInCart = await securePaymentComponent.numberOfProductsInCart();
+				return assert.strictEqual(
+					numberOfProductsInCart,
+					1,
+					"The cart doesn't contain the expected number of products"
+				);
+			}
+		);
+
+		step(
+			'Can then see the secure payment page with the expected currency in the cart',
+			async function() {
+				const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+				if ( driverManager.currentScreenSize() === 'desktop' ) {
+					const totalShown = await securePaymentComponent.cartTotalDisplayed();
+					assert.strictEqual(
+						totalShown.indexOf( expectedCurrencySymbol ),
+						0,
+						`The cart total '${ totalShown }' does not begin with '${ expectedCurrencySymbol }'`
+					);
+				}
+				const paymentButtonText = await securePaymentComponent.paymentButtonText();
+				return assert(
+					paymentButtonText.includes( expectedCurrencySymbol ),
+					`The payment button text '${ paymentButtonText }' does not contain the expected currency symbol: '${ expectedCurrencySymbol }'`
+				);
+			}
+		);
+
+		step( 'Can Correctly Apply Coupon discount', async function() {
+			const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+			await securePaymentComponent.toggleCartSummary();
+			originalCartAmount = await securePaymentComponent.cartTotalAmount();
+
+			await securePaymentComponent.enterCouponCode( dataHelper.getTestCouponCode() );
+
+			let newCartAmount = await securePaymentComponent.cartTotalAmount();
+			let expectedCartAmount = parseFloat( ( originalCartAmount * 0.99 ).toFixed( 2 ) );
+			assert.strictEqual( newCartAmount, expectedCartAmount, 'Coupon not applied properly' );
+		} );
+
+		step( 'Can enter and submit test payment details', async function() {
```

Comment:
Since we're not using a sandboxed mode, we can remove the rest of this test and just let the test finish at the coupon in the cart step